R/derive_locf_records.R

```diff
@@ -69,7 +69,7 @@
 #' advs <- tribble(
 #'   ~STUDYID,  ~USUBJID,      ~PARAMCD, ~PARAMN, ~AVAL, ~AVISITN, ~AVISIT,
 #'   "CDISC01", "01-701-1015", "PULSE",        1,    61,        0, "BASELINE",
-#'   "CDISC01", "01-701-1015", "PULSE",        1,    60,        2, "WEEK 6",
+#'   "CDISC01", "01-701-1015", "PULSE",        1,    60,        2, "WEEK 2",
```

Comment:
I think we should change to the following to be inline with the expected visits.
```suggestion
#'   "CDISC01", "01-701-1015", "PULSE",        1,    60,        6, "WEEK 6",
```

Reply:
Hi @bundfussr do you think we should put VISIT = week 6 and AVISITN= 6  instead of AVISIT = week2 and AVISITN = 2?

Reply:
Yes, because for PULSE the visits Baseline and Week 6 are expeted.
```
# A dataset with all the combinations of PARAMCD, PARAM, AVISIT, AVISITN, ... which are expected.
advs_expected_obsv <- tribble(
  ~PARAMCD, ~AVISITN, ~AVISIT,
  "PULSE",         0, "BASELINE",
  "PULSE",         6, "WEEK 6",
```

Reply:
Hi @janardhanswami can you please update and we shall get this merged in!!  

We discussed as a team and don't think admiral should check 1-1 mappings within our functions.  It might causes too many issues with dirty data among other things.  

We are going to provide some recommendations on how to check 1-1 mappings as a pre-processing or post-processing step here: https://github.com/pharmaverse/admiral/issues/2395.  It would be great to get your review/opinion on new vignettes.  If you are interested just tag yourself in the issue so we are reminded to flag you in the review.  

Your finding generated some really great discussion.  Thanks again!

Reply:
Hi @bms63 I'll update that  and thanks for the information.

Reply:
Hi @bms63  I've just updated the example by removing the typo, let me know if we can merge or should I have to do anything else?